dist/secure-handlebars.js

```diff
@@ -9744,7 +9832,202 @@ module.exports = ContextParserHandlebars;
 })();
 
 }).call(this,require('_process'))
-},{"./handlebars-utils.js":40,"./strict-context-parser.js":42,"_process":8,"context-parser":1,"debug":3}],40:[function(require,module,exports){
+},{"./css-parser.js":41,"./handlebars-utils.js":42,"./strict-context-parser.js":44,"_process":8,"context-parser":1,"debug":3}],41:[function(require,module,exports){
+/* 
+Copyright (c) 2015, Yahoo Inc. All rights reserved.
+Copyrights licensed under the New BSD License.
+See the accompanying LICENSE file for terms.
+
+Authors: Nera Liu <neraliu@yahoo-inc.com>
+         Albert Yu <albertyu@yahoo-inc.com>
+         Adonis Fung <adon@yahoo-inc.com>
+*/
+(function () {
+"use strict";
+
+/////////////////////////////////////////////////////
+//
+// @module CSSParser
+// 
+/////////////////////////////////////////////////////
+var CSSParser = {};
+
+/*
+'&#0*8722;?|&#[xX]0*2212;?|&minus;|&#0*8208;?|&#[xX]0*2010;?|&dash;|&hyphen;'		// minus & dash
+'&#0*92;?|&#[xX]0*5[cC];?|&bsol;'		 	// \
+'&#0*95;?|&#[xX]0*5[fF];?|&lowbar;|&UnderBar;'		// _
+'&#0*58;?|&#[xX]0*3[aA];?|&colon;'      		// colon
+'&#0*59;?|&#[xX]0*3[bB];?|&semi;'       		// semicolon
+'&#0*40;?|&#[xX]0*28;?|&lpar;'          		// (
+'&#0*41;?|&#[xX]0*29;?|&rpar;'          		// )
+'&#0*34;?|&#[xX]0*22;?|&quot;|&QUOT;'			// "
+'&#0*39;?|&#[xX]0*27;?|&apos;'				// '
+'&#0*47;?|&#[xX]0*2[fF];?|&sol;'			// /
+'&#0*44;?|&#[xX]0*2[cC];?|&comma;'			// ,
+'&#0*43;?|&#[xX]0*2[bB];?|&plus;'			// +
+'&#0*37;?|&#[xX]0*25;?|&percnt;'			// percentage
+'&#0*35;?|&#[xX]0*23;?|&num;'				// hash
+'&#0*33;?|&#[xX]0*21;?|&excl;'		 		// !
+'&#0*42;?|&#[xX]0*2[aA];?|&ast;|&midast;'		// *
+'&#0*32;?|&#[xX]0*20;?|&#0*9;?|&#[xX]0*9;?|&Tab;|&#0*10;?|&#[xX]0*[aA];?|&NewLine;|&#0*12;?|&#[xX]0*[cC];?|&#0*13;?|&#[xX]0*[dD];?|\t|\r|\n|\f'; // space,\t,\r,\n,\f
+
+note: the decoding order is important, as we are using regExp not real parsing. 
+for example, we need to replace \ before space, as &#0*92 needs to be checked before &#0*9.
+the correct decoding order is to match as greedy as possible first, that is matching N digit before N-1 digit.
+
+*/
+CSSParser.cssReplaceChar = [ '-', '\\', '_', ':', ';', '(', ')', '"', '\'', '\/', ',', '+', '%', '#', '!', '*', ' ' ];
+CSSParser.reCss = [
+    /&#0*8722;?|&#[xX]0*2212;?|&minus;|&#0*8208;?|&#[xX]0*2010;?|&dash;|&hyphen;/g,
```

Comment:
`'-'.charCodeAt(0)` returns 45 (decimal). that hyphen symbol is copied from you `reExpr`. is 45 what you want to catch instead?